src/main/java/seedu/address/model/room/RoomNumberOrTagsContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,29 @@
+package seedu.address.model.room;
+
+import java.util.List;
+import java.util.function.Predicate;
+/**
+ * Tests that a {@code Room}'s {@code RoomNumber} matches any of the keywords given.
+ */
+public class RoomNumberOrTagsContainsKeywordsPredicate implements Predicate<Room> {
+    private final List<String> keywords;
+
+    public RoomNumberOrTagsContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
```

Comment:
Consider adding JavaDocs to this because it is the key part of a predicate.

Reply:
Done